utils/get_supported_scylla_base_versions.py

```diff
@@ -0,0 +1,137 @@
+        to opensource. This function is used to get the base versions list which will be used in the upgrade test.
+
+        @supported_version: all scylla release versions, the base versions will be filtered out from the supported_version
+        """
+        oos_base_version = []
+        ent_base_version = []
+
+        oos_release_list = [i for i in supported_versions if not is_enterprise(i)]
+        ent_release_list = [i for i in supported_versions if is_enterprise(i)]
+
+        # The major version of unstable scylla, eg: 4.6.dev, 4.5, 2021.2.dev, 2021.1
+        version = self.scylla_version
+        product = self.product
+
+        if product == 'scylla-enterprise':
+            if version in supported_versions:
+                # The dest version is a released enterprise version
+                idx = ent_release_list.index(version)
+                oos_base_version.append(supported_src_oos.get(version))
+                if idx == 0:
+                    # The dest version is a minor release of latest enterprise release
+                    ent_base_version.append(version)
+                else:
+                    # Choose the last two releases as upgrade base
+                    ent_base_version += ent_release_list[idx-1:][:2]
+            elif version == 'enterprise' or parse_version(version) > parse_version(ent_release_list[0]):
+                oos_base_version.append(ent_release_list[-1])
+            elif re.match(r'\d+.\d+', version) and parse_version(version) >= parse_version(ent_release_list[0]):
+                oos_base_version.append(oos_release_list[-1])
+                ent_base_version += ent_release_list[-2:]
+        elif product == 'scylla':
+            if version in supported_versions:
+                # The dest version is a released opensource version
+                idx = oos_release_list.index(version)
+                if idx == 0:
+                    # The dest version is a minor release of latest enterprise release
+                    oos_base_version.append(version)
+                else:
+                    # Choose the last two releases as upgrade base
+                    oos_base_version += oos_release_list[idx-1:][:2]
+            elif version == 'master' or parse_version(version) > parse_version(oos_release_list[0]):
+                oos_base_version.append(oos_release_list[-1])
+            elif re.match(r'\d+.\d+', version) and parse_version(version) < parse_version(oos_release_list[0]):
+                # If dest version is smaller than the first supported opensource release, it might be an invalid dest version
+                oos_base_version.append(oos_release_list[-1])
+        else:
+            raise ValueError("Unsupported product %s" % product)
+
+        # Filter out unsupported version
+        oos_base_version = [i for i in oos_base_version if i in supported_versions]
+        ent_base_version = [i for i in ent_base_version if i in supported_versions]
```

Comment:
Small nit: "i" is used as step item for "Integer"s. "s" for "String"s. But also, instead of string we may use other. In currect case it may be "version" or just "v".

Typo. We have OSS, not OOS. -> Open Source Scylla.

Reply:
fixed.